src/systemcmds/ver/ver.cpp

```diff
@@ -148,79 +148,86 @@ extern "C" __EXPORT int ver_main(int argc, char *argv[])
 					snprintf(rb, sizeof(rb), "0x%0" STRINGIFY(HW_INFO_REV_DIGITS) "X", r);
 				}
 
-				printf("HW type: %s\n", strlen(px4_board_sub_type()) ? px4_board_sub_type() : "NA");
-				printf("HW version: %s\n", vb);
-				printf("HW revision: %s\n", rb);
+				PX4_INFO_RAW("HW type: %s\n", strlen(px4_board_sub_type()) ? px4_board_sub_type() : "NA");
+				PX4_INFO_RAW("HW version: %s\n", vb);
+				PX4_INFO_RAW("HW revision: %s\n", rb);
 #endif
 				ret = 0;
 
 			}
 
 			if (show_all || !strncmp(argv[1], sz_ver_git_str, sizeof(sz_ver_git_str))) {
-				printf("PX4 git-hash: %s\n", px4_firmware_version_string());
+				PX4_INFO_RAW("PX4 git-hash: %s\n", px4_firmware_version_string());
 				unsigned fwver = px4_firmware_version();
 				unsigned major = (fwver >> (8 * 3)) & 0xFF;
 				unsigned minor = (fwver >> (8 * 2)) & 0xFF;
 				unsigned patch = (fwver >> (8 * 1)) & 0xFF;
 				unsigned type = (fwver >> (8 * 0)) & 0xFF;
 
 				if (type == 255) {
-					printf("PX4 version: Release %u.%u.%u (%u)\n", major, minor, patch, fwver);
+					PX4_INFO_RAW("PX4 version: Release %u.%u.%u (%u)\n", major, minor, patch, fwver);
 
 				} else {
-					printf("PX4 version: %u.%u.%u %x (%u)\n", major, minor, patch, type, fwver);
+					PX4_INFO_RAW("PX4 version: %u.%u.%u %x (%u)\n", major, minor, patch, type, fwver);
 				}
 
 				if (show_all) {
 					const char *git_branch = px4_firmware_git_branch();
 
 					if (git_branch && git_branch[0]) {
-						printf("PX4 git-branch: %s\n", git_branch);
+						PX4_INFO_RAW("PX4 git-branch: %s\n", git_branch);
 					}
 				}
 
+				fwver = px4_firmware_vendor_version();
```

Comment:
I would only print it if it's non-zero

Reply:
Okay, I updated it to only print the vendor version when it is non-zero.